File: FavoriteSequence.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);
        for (auto &x:arr)
            cin >> x;
        vector<int> sol;
        int i=0, j=n-1;
        bool back = false;
        while (n--) {
            if (!back) {
                sol.push_back(arr[i]);
                i++;
                back = true;
            }
            else {
                sol.push_back(arr[j]);
                j--;
                back = false;
            }
        }
        for (auto &x:sol)
            cout << x << " ";
        cout << endl;
    }

    return 0;
}

File: Python/FavoriteSequence.py
def main():
    t = int(input())
    while t:
        t -= 1
        n = int(input())
        arr = [int(input()) for i in range(n)]
        sol = []
        i = 0
        j = n-1
        back = False
        while n:
            n -= 1
            if ~back:
                sol.append(arr[i])
                i += 1
                back = True
            else:
                sol.append(arr[j])
                j -= 1
                back = False
        for i in sol:
            print(x, end=" ")
        print()


if __name__ == '__main__':
    main()
